src/ansi-c/c_typecheck_expr.cpp

```diff
@@ -966,18 +973,20 @@ void c_typecheck_baset::typecheck_expr_sizeof(exprt &expr)
   {
     // This is a gcc extension.
     // https://gcc.gnu.org/onlinedocs/gcc-4.8.0/gcc/Pointer-Arith.html
-    new_expr = size_of_expr(char_type(), *this);
+    new_expr = from_integer(1, size_type());
```

Comment:
This appears to be a correct, but unrelated change.

Reply:
It's getting rid of one use of `size_of_expr`, which would now entail the use of `optionalt`